src/engine/plugins/trip.cpp

```diff
@@ -187,56 +214,48 @@ Status TripPlugin::HandleRequest(const std::shared_ptr<const datafacade::BaseDat
     BOOST_ASSERT_MSG(result_table.size() == number_of_locations * number_of_locations,
                      "Distance Table has wrong size");
 
-    // get scc components
-    SCC_Component scc = SplitUnaccessibleLocations(number_of_locations, result_table);
-
-    std::vector<std::vector<NodeID>> trips;
-    trips.reserve(scc.GetNumberOfComponents());
-    // run Trip computation for every SCC
-    for (std::size_t k = 0; k < scc.GetNumberOfComponents(); ++k)
+    if (!IsStronglyConnectedComponent(result_table))
     {
-        const auto component_size = scc.range[k + 1] - scc.range[k];
-
-        BOOST_ASSERT_MSG(component_size > 0, "invalid component size");
-
-        std::vector<NodeID> scc_route;
-        auto route_begin = std::begin(scc.component) + scc.range[k];
-        auto route_end = std::begin(scc.component) + scc.range[k + 1];
-
-        if (component_size > 1)
-        {
+        return Error("NoTrips", "No trip visiting all destinations possible.", json_result);
+    }
 
-            if (component_size < BF_MAX_FEASABLE)
-            {
-                scc_route =
-                    trip::BruteForceTrip(route_begin, route_end, number_of_locations, result_table);
-            }
-            else
-            {
-                scc_route = trip::FarthestInsertionTrip(
-                    route_begin, route_end, number_of_locations, result_table);
-            }
-        }
-        else
-        {
-            scc_route = std::vector<NodeID>(route_begin, route_end);
-        }
+    if (fixed_start_and_end)
+    {
+        ManipulateTableForFSE(source_id, destination_id, result_table);
+    }
 
-        trips.push_back(std::move(scc_route));
+    std::vector<NodeID> trip;
+    trip.reserve(number_of_locations);
+    // get an optimized order in which the destinations should be visited
+    if (number_of_locations < BF_MAX_FEASABLE)
+    {
+        trip = trip::BruteForceTrip(number_of_locations, result_table);
     }
-    if (trips.empty())
+    else
     {
-        return Error("NoTrips", "Cannot find trips", json_result);
+        trip = trip::FarthestInsertionTrip(number_of_locations, result_table);
     }
 
-    // compute all round trip routes
-    std::vector<InternalRouteResult> routes;
-    routes.reserve(trips.size());
-    for (const auto &trip : trips)
+    // rotate result such that roundtrip starts at node with index 0
+    if (!(fixed_end && !fixed_start_and_end))
```

Comment:
`!fixed_end || fixed_start` is the same as `!fixed_end OR
fixed_start OR fixed_start && fixed_end`, because `fixed_start OR fixed_start && fixed_end` is equal to `fixed_start`